pylint/config/option.py

```diff
@@ -194,6 +194,8 @@ def _check_choice(self):
     optparse.Option.CHECK_METHODS[2] = _check_choice  # type: ignore[index]
 
     def process(self, opt, value, values, parser):
+        if self.callback and self.callback.__module__ == "pylint.lint.run":
```

Comment:
This makes it so that whenever we encounter a callback function defined in our own `Run` file we don't execute. Other callback defines by plugins are still being executed.

This is not necessarily a problem: their callbacks are simply "executed" by `optparse`. I thought it would be a good milestone for us to let `argparse` handle this.